src/flow/factory.js

```diff
@@ -0,0 +1,124 @@
+/**
```

Comment:
This filename doesn't tell me much about what this is doing. I had to guess. Can we put something in the filename and preferably the folder name about this being specifically about turning Flow types into Abstract Objects?

Not any other Flow support such as preserving annotation and adding more while cross-compiling.

Reply:
Agreed, naming things is hard. Will change the name, but leave it in the folder as it's related to Flow.

Reply:
`abstractObjectFactories.js`?